music21/spanner.py

```diff
@@ -610,6 +613,67 @@ class _SpannerRef(t.TypedDict):
     spanner: 'Spanner'
     className: str
 
+class SpannerAnchor(base.Music21Object):
+    '''
+    A simple Music21Object that can be used to define the beginning or end
+    of a Spanner, in the place of a GeneralNote.
+
+    This is useful for (e.g.) a Crescendo that ends partway through a
+    note (e.g. in a violin part).
+
+    Here's an example of a whole note that has a Crescendo for the first
+    half of the note, and a Diminuendo for the second half of the note.
+
+    >>> score = stream.Score()
+    >>> part = stream.Part()
+    >>> score.insert(0, part)
+    >>> measure = stream.Measure()
+    >>> part.insert(0, measure)
+    >>> voice = stream.Voice()
+    >>> measure.insert(0, voice)
+
+    >>> n = note.Note('C', quarterLength = 4)
+    >>> sa1 = spanner.SpannerAnchor()
+    >>> sa2 = spanner.SpannerAnchor()
+    >>> voice.insert(0, n)
+    >>> voice.insert(2, sa1)
+    >>> voice.insert(4, sa2)
+    >>> cresc = dynamics.Crescendo(n, sa1)   # cresc from n to sa1
+    >>> dim = dynamics.Diminuendo(sa1, sa2)  # dim from sa1 to sa2
+    >>> score.append((cresc, dim))
+
+    SpannerAnchors aways have a duration of 0, and if any attempt is made to
+    change this, TypeError will be raised.
+    '''
+    def __init__(self, *spannedElements, **keywords):
+        if 'duration' in keywords or 'quarterLength' in keywords:
+            raise TypeError(
+                'SpannerAnchor cannot be initialized with a duration/quarterLength.'
+            )
+
+        super().__init__(*spannedElements, **keywords)
+        self._duration = duration.FrozenDuration(quarterLength=0)
+
+    @property
+    def duration(self) -> duration.Duration:
+        d_out = self._duration
+        if t.TYPE_CHECKING:
+            assert d_out is not None
+        return d_out
+
+    @duration.setter
+    def duration(self, durationObj: duration.Duration):
+        raise TypeError('SpannerAnchor has an immutable zero duration.')
+
+    @property
+    def quarterLength(self) -> OffsetQL:
+        return self.duration.quarterLength
+
+    @quarterLength.setter
+    def quarterLength(self, value: OffsetQLIn):
+        raise TypeError('SpannerAnchor has an immutable zero quarterLength.')
```

Comment:
Seems unnecessary: commenting it out still gives:

```python
>>> sa = SpannerAnchor()
>>> sa.quarterLength = 3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/.../music21/music21/base.py", line 826, in quarterLength
    self.duration.quarterLength = value
  File "/Users/.../music21/music21/common/objects.py", line 294, in __setattr__
    self._check_init(key)
  File "/Users/.../music21/music21/common/objects.py", line 291, in _check_init
    raise TypeError(f'This {self.__class__.__name__} instance is immutable.')
TypeError: This FrozenDuration instance is immutable.
```

Reply:
Done.